synapse/handlers/identity.py

```diff
@@ -447,6 +447,34 @@ def requestMsisdnToken(
             logger.info("Proxied requestToken failed: %r", e)
             raise e.to_synapse_error()
 
+    @defer.inlineCallbacks
+    def proxy_msisdn_submit_token(self, id_server, client_secret, sid, token):
+        """Proxy a POST submitToken request to an identity server for verification purposes
+
+        Args:
+            id_server (str): The identity server URL to contact
+
+            client_secret (str): Secret provided by the client
+
+            sid (str): The ID of the session
+
+            token (str): The verification token
+
+        Raises:
+            SynapseError: If we failed to contact the identity server
+
+        Returns:
+            The response dict from the identity server
+        """
+        body = {"client_secret": client_secret, "sid": sid, "token": token}
+
+        return (
+            yield self.http_client.post_json_get_json(
+                id_server + "/_matrix/identity/api/v1/validate/msisdn/requestToken",
```

Comment:
Shouldn't that be using the v2 API (or use the v2 API and fallback to the v1 one)?

Reply:
Also, shouldn't that be `submitToken` and not `requestToken`?

Reply:
Yes, that should be submitToken.

Reply:
@babolivier nah, were using v1 endpoints for delegates. There's no need for a user to authenticate against them.

Reply:
```suggestion
                id_server + "/_matrix/identity/api/v1/validate/msisdn/submitToken",
```